tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,282 @@
+
+        interrupt_test_streams = {'collects',
+                                  'customers',
+                                  'events',
+                                  'metafields',
+                                  'orders',
+                                  'products',
+                                  'transactions'}
+
+        # Select all streams and no fields within streams
+        found_catalogs = menagerie.get_catalogs(conn_id)
+
+        # Our test data sets for Shopify do not have any abandoned_checkouts
+        our_catalogs = [catalog for catalog in found_catalogs if
+                        catalog.get('tap_stream_id') in interrupt_test_streams]
+        self.select_all_streams_and_fields(conn_id, our_catalogs, select_all_fields=True)
+
+        #################################
+        # Run first sync
+        #################################
+
+        first_sync_record_count = self.run_sync(conn_id)
+        first_sync_state = menagerie.get_state(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+        first_sync_order = runner.get_stream_sync_order_from_target()
+
+        LOGGER.info(f"First sync stream order: {first_sync_order}")
+
+        # verify that the sync only sent records to the target for selected streams (catalogs)
+        self.assertEqual(set(first_sync_record_count.keys()), interrupt_test_streams)
+
+        # BUG:TDL-17087 : State has additional values which are not streams
+        # Need to remove additional values from bookmark value
+        extra_stuff = {'transaction_orders',
+                       'metafield_products',
+                       'refund_orders',
+                       'product_variants'}
+
+        for keys in list(first_sync_state['bookmarks'].keys()):
+            if keys in extra_stuff:
+                first_sync_state['bookmarks'].pop(keys)
+
+        ################################
+        # Update State between Syncs
+        ################################
+
+        # hardcoding the updated state to ensure atleast 1 record in resuming (2nd) sync.
+        # values have been provided after reviewing the max bookmark value for each of the streams
+        new_state = {'bookmarks':
+                     {'currently_sync_stream': 'customers',
+                      'collects': {'updated_at': '2023-01-01T09:08:28.000000Z'},
+                      'customers': {'updated_at': '2023-03-28T18:53:28.000000Z'},
+                      'events': {'created_at': '2023-01-22T05:05:53.000000Z'},
+                      #'metafields': {'updated_at': '2023-01-07T21:18:05.000000Z'},
+                      #'orders': {'updated_at': '2023-01-20T05:09:01.000000Z'},
+                      #'products': {'updated_at': '2023-01-20T05:10:05.000000Z'},
+                      'transactions': {'created_at': '2022-06-26T00:06:38-04:00'}
+                      }}
+
+        currently_syncing_stream = new_state['bookmarks']['currently_sync_stream']
```

Comment:
Nitpick: It's clearer if you define `currently_syncing_stream` and then use it in the `new_state`

```suggestion
        currently_syncing_stream = 'customers'

        new_state = {'bookmarks':
                     {'currently_sync_stream': currently_syncing_stream,
                      'collects': {'updated_at': '2023-01-01T09:08:28.000000Z'},
                      'customers': {'updated_at': '2023-03-28T18:53:28.000000Z'},
                      'events': {'created_at': '2023-01-22T05:05:53.000000Z'},
                      #'metafields': {'updated_at': '2023-01-07T21:18:05.000000Z'},
                      #'orders': {'updated_at': '2023-01-20T05:09:01.000000Z'},
                      #'products': {'updated_at': '2023-01-20T05:10:05.000000Z'},
                      'transactions': {'created_at': '2022-06-26T00:06:38-04:00'}
                      }}
```